dali/base/dameta.cpp

```diff
@@ -38,6 +44,43 @@ IPropertyTree * queryStoragePlane(const char * name)
     return global.queryPropTree(xpath);
 }
 
+
+// Expand indirect hostGroups so each hostGroups has an expanded list of host names
+void normalizeHostGroups()
+{
+    Owned<IPropertyTreeIterator> hostGroupIter = queryGlobalConfig().getElements("storage/hostGroups");
+    //Process the groups in order - so that multiple levels of indirection are supported
+    ForEach (*hostGroupIter)
+    {
+        IPropertyTree & cur = hostGroupIter->query();
+        if (!cur.hasProp("hosts"))
+        {
+            const char * name = cur.queryProp("@name");
+            const char * baseGroup = cur.queryProp("@hostGroup");
+            IPropertyTree * match = queryHostGroup(baseGroup, true);
+            StringArray hosts;
+            Owned<IPropertyTreeIterator> hostIter = match->getElements("hosts");
+            ForEach (*hostIter)
+                hosts.append(hostIter->query().queryProp(nullptr));
+
+            if (hosts.ordinality() == 0)
+                throw makeStringExceptionV(-1, "Host group %s contains no hosts", baseGroup);
+
+            unsigned numHosts = cur.getPropInt("@count", hosts.ordinality());
+            unsigned offset = cur.getPropInt("@offset");
+            if (offset + numHosts > hosts.ordinality())
+                throw makeStringExceptionV(-1, "Group %s extends past the end of the base group %s", name, baseGroup);
+
+            unsigned delta = cur.getPropInt("@delta");
+            for (unsigned i=0; i < numHosts; i++)
+            {
+                unsigned baseIndex = offset + (i + delta) % numHosts;
```

Comment:
related to comment re. simpler if just 'offset' and 'count' in schema..
this would work without delta, if planes used different offsets, with count and wrapping was implied.

Reply:
see comment above - it is needed.